homeassistant/components/mobile_app/notify.py

```diff
@@ -0,0 +1,131 @@
+"""Support for mobile_app push notifications."""
+import asyncio
+from datetime import datetime, timezone
+import logging
+
+import async_timeout
+
+from homeassistant.components.notify import (
+    ATTR_DATA, ATTR_MESSAGE, ATTR_TARGET, ATTR_TITLE, ATTR_TITLE_DEFAULT,
+    BaseNotificationService)
+from homeassistant.components.mobile_app.const import (
+    ATTR_APP_DATA, ATTR_APP_ID, ATTR_APP_VERSION, ATTR_DEVICE_NAME,
+    ATTR_OS_VERSION, ATTR_PUSH_TOKEN, ATTR_PUSH_URL, DATA_CONFIG_ENTRIES,
+    DOMAIN)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['mobile_app']
+
+
+def push_registrations(hass):
+    """Return a dictionary of push enabled registrations."""
+    targets = {}
+    for webhook_id, entry in hass.data[DOMAIN][DATA_CONFIG_ENTRIES].items():
+        data = entry.data
+        app_data = data[ATTR_APP_DATA]
+        if ATTR_PUSH_TOKEN in app_data and ATTR_PUSH_URL in app_data:
+            device_name = data[ATTR_DEVICE_NAME]
```

Comment:
I would put in a check that if a device name is overridden, we give a warning and not override it.

Reply:
@balloob Looked through a bunch of config entry stuff and couldn't find an example, got a handy one?

Reply:
```python
if device_name in targets:
    _LOGGER.warning("Found duplicate device name %s", device_name)
    continue
```

Reply:
oh duh, i thought you meant if overriden in the UI